core/server/src/log/logger.rs

```diff
@@ -253,8 +255,48 @@ impl Logging {
             let base_directory = PathBuf::from(base_directory);
             let logs_subdirectory = PathBuf::from(config.path.clone());
             let logs_path = base_directory.join(logs_subdirectory.clone());
-            let file_appender =
-                tracing_appender::rolling::hourly(logs_path.clone(), IGGY_LOG_FILE_PREFIX);
+
+            if let Err(e) = std::fs::create_dir_all(&logs_path) {
+                tracing::warn!("Failed to create logs directory {:?}: {}", logs_path, e);
+                return Err(LogError::FileReloadFailure);
+            }
+
+            // Check available disk space (at least 10MB)
+            let min_disk_space: u64 = 10 * 1024 * 1024; // 10MB
```

Comment:
Let's put this value into const on top, something like `const MIN_DISK_SPACE_MB: u64 = 10`